src/EditorFeatures/Test2/Rename/InlineRenameTests.vb

```diff
@@ -1558,5 +1568,163 @@ End Class
+                Assert.True(session.AllowFileRename)
+            End Using
+        End Sub
+
+        <WpfFact>
+        <Trait(Traits.Feature, Traits.Features.Rename)>
+        Public Sub VerifyRenameAllowedWithMultipleTypesOnMatchingName()
+            Using workspace = CreateWorkspaceWithWaiter(
+                    <Workspace>
+                        <Project Language="C#" CommonReferences="true">
+                            <Document>
+                                class [|$$Test1|]
+                                {
+                                    void Blah()
+                                    {
+                                    }
+                                }
+
+                                class Test2
+                                {
+                                }
+                            </Document>
+                        </Project>
+                    </Workspace>)
+
+                Dim session = StartSession(workspace)
+
+                Assert.True(session.KindSupportsFileRename)
+                Assert.True(session.AllowFileRename)
+            End Using
+        End Sub
+
+        <WpfFact>
+        <Trait(Traits.Feature, Traits.Features.Rename)>
+        Public Sub VerifyNoRenameAllowedWithMultipleTypes()
+            Using workspace = CreateWorkspaceWithWaiter(
+                    <Workspace>
+                        <Project Language="C#" CommonReferences="true">
+                            <Document>
+                                class [|$$Goo|]
+                                {
+                                    void Blah()
+                                    {
+                                    }
+                                }
+
+                                class Test1
+                                {
+                                }
+                            </Document>
+                        </Project>
+                    </Workspace>)
+
+                Dim session = StartSession(workspace)
+
+                Assert.True(session.KindSupportsFileRename)
+                Assert.False(session.AllowFileRename)
+            End Using
+        End Sub
+
+        <WpfFact>
+        <Trait(Traits.Feature, Traits.Features.Rename)>
+        Public Sub VerifyEnumKindSupportsRename()
+            Using workspace = CreateWorkspaceWithWaiter(
+                    <Workspace>
+                        <Project Language="C#" CommonReferences="true">
+                            <Document>
+                                enum [|$$Goo|]]
+                                {
+                                    One,
+                                    Two,
+                                    Three
+                                }
+                            </Document>
+                        </Project>
+                    </Workspace>)
+
+                Dim session = StartSession(workspace)
+
+                Assert.True(session.KindSupportsFileRename)
```

Comment:
Do we have a test asserting false for the non-Type case?

Reply:
*Looks at diff sitting on computer* Yes

Coming soon :)